lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -3,10 +3,23 @@
 module Reports
   class SiegeReport
     def initialize(building:)
+      @building = building
     end
 
     def call
-      raise NotImprementedYet
+      daily_supplies = 10
+      if @building.warriors.alive.any?
+        @building.warriors.alive.each do |warrior|
+          daily_supplies += if warrior.type == 'Warriors::Hussar'
```

Comment:
Ładniej by to chyba wyglądało, jeśli Warrior miałby zaimplementowaną metodę/atrybut np. `supply_burn_rate`, który zwraca 1 jeśli warrior nie ma konia i 2 jeśli go ma, wtedy nie trzeba by myśleć nad takimi ifami w tym serwisie :)